plugins/nanospell/plugin.js

```diff
@@ -942,7 +953,7 @@
 
 						if (retval) {
 							var currentData = self.clearAllSpellCheckingSpansFromString(this.getSnapshot()),
-								prevData = self.clearAllSpellCheckingSpansFromString(this._.previousValue);
+								prevData = this._.previousValue;
```

Comment:
This is an override for editor's checkDirty.  

this._.previousValue gets set in resetDirty, which we also override below.  the original code is http://docs.ckeditor.com/source/editor.html#CKEDITOR-editor-method-resetDirty

It's not super well documented, but the official scayt uses a similar strategy for clearing spans when converting to and from source.